lib/folio_client/inventory.rb

```diff
@@ -22,5 +22,21 @@ def fetch_hrid(barcode:)
       result = client.get("/inventory/instances/#{instance_uuid}")
       result.dig("hrid")
     end
+
+    # @param hrid [String] folio instance HRID
+    # @param status_code [String] id for the "cat" instance status code
+    def cataloged?(hrid:, status_code:)
+      # get the instance record and its statusId
+      instance = client.get("/inventory/instances", {query: "hrid==#{hrid}"})
+      raise StandardError, "No matching instance found for #{hrid}" if instance["totalRecords"] == 0
```

Comment:
There is no "not found" error?

Reply:
well, that's a stupid interpretation of HTTP.  Yay.

Reply:
Actually, let me clarify. Why not raise a `FolioClient::UnexpectedResponse::ResourceNotFound`?

Reply:
Ah, thanks. I see we're using it currently for a 404 ResourceNotFound which had me thinking that was for the query being invalid or the API endpoint being wrong. If it sounds right to use that error here, too though--which you're suggesting, then I'll change it.